docs/modules/ROOT/pages/extensions/freemarker.adoc

```diff
@@ -0,0 +1,53 @@
+[[freemarker]]
+= Freemarker Extension
+
+*Since Camel Quarkus 0.3.0*
+
+The Freemarker extension component allows processing a message using a FreeMarker template. This can be ideal when using Templating to generate responses for requests.
+
+Maven users will need to add the following dependency to their `pom.xml` for this extension.
+
+[source,xml]
+------------------------------------------------------------
+<dependency>
+    <groupId>org.apache.camel.quarkus</groupId>
+    <artifactId>camel-quarkus-freemarker</artifactId>
+</dependency>
+------------------------------------------------------------
+
+== Usage
+
+The extension provides support for the Camel https://camel.apache.org/components/latest/freemarker-component.html[Freemarker Component].
+
+=== Configuration
+
+The Camel Freemarker extension exposes the following configuration properties.
+
+[cols="80,.^10,.^10"]
+|===
+|Configuration property |Type |Default
+
+|`quarkus.camel.freemarker.locations`
```

Comment:
Could we perhaps remove the locations option now that we have includes and excludes? The includes option seems to offer a superset of `locations` functionality. The default can be changed to `freemarker/templates/**` and moved to the includes option too. WDYT?